File: Bubble-Sort/BubbleSort.rb
#
#  run ruby BubbleSort.rb
#

def bubbleSort array
    loop do
        swapped = false

        (array.length-1).times do |i|
            if array[i] > array[i+1]
                array[i], array[i+1] = array[i+1], array[i]
                swapped = true
            end
        end

        break if not swapped
    end
    array
end

array = [2,4,1,5,7,8, -5, 0]

p bubbleSort array

File: Bubble-Sort/BubbleSort.ts
function bubbleSort(array:number[]):number[]{
    array = array.slice();
    console.log(array);
    for( let i=0;i<array.length;i++){ 
      for(let j=0;j<array.length-1;j++){
        if(array[j] > array[j+1]){
          [array[j], array[j+1]] =[array[j+1],array[j]];
          console.log(array);
        }
      }
    }
    
    return array;
  }
  
  bubbleSort([5,2,8,1,3,2,4,54,12,12]);